core/serialization/deserializer/montage-deserializer.js

```diff
@@ -27,39 +23,62 @@ var MontageDeserializer = exports.MontageDeserializer = Montage.specialize({
     },
 
     init: {
-        value: function (serializationString, _require, objectRequires, locationId, moduleContexts) {
-            this._serializationString = serializationString;
-            moduleContexts = moduleContexts || new Map();
-            this._interpreter = new MontageInterpreter().init(_require,
-                new MontageReviver().init(_require, objectRequires, locationId, moduleContexts));
+        value: function (serialization, _require, objectRequires, locationId) {
+            if (typeof serialization === "string") {
+                this._serializationString = serialization;
+            } else {
+                this._serializationString = JSON.stringify(serialization);
+            }
+            this._require = _require;
+            this._locationId = locationId;
+            this._reviver = new MontageReviver().init(_require, objectRequires,
+                this._childConstructor.bind(this));
```

Comment:
This will create a new function everytime a deserializer is created, can't we use call() or pass this as an extra argument instead?